File: climb-log-flutter/lib/climasys/api/models/facility.dart
class Facility {
  final int id;
  final String? iconName;
  final String name;

  Facility({
    required this.id,
    required this.iconName,
    required this.name,
  });

  factory Facility.fromJson(Map<String, dynamic> json) {
    return Facility(
      id: json['id'],
      iconName: json['iconName'],
      name: json['name'],
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'iconName': iconName,
    'name': name,
  };
}

File: ClimbLog.Api/ClimbLog.Domain/Models/Entities/Facility.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClimbLog.Domain.Models.Entities;

public class Facility
{
    public int Id { get; set; }
    public string? IconName { get; set; }
    public required string Name { get; set; }
}
